blockdag/blockwindow.go

```diff
@@ -53,13 +54,37 @@ func (window blockWindow) minMaxTimestamps() (min, max int64) {
 	return
 }
 
+var blockWindowBigIntPool = sync.Pool{
+	New: func() interface{} {
+		return big.NewInt(0)
+	},
+}
+
+func acquireBigInt() *big.Int {
+	return blockWindowBigIntPool.Get().(*big.Int)
+}
+
+func releaseBigInt(toRelease *big.Int) {
+	toRelease.SetInt64(0)
+	blockWindowBigIntPool.Put(toRelease)
+}
+
```

Comment:
If it's a general big int pool, call it with a generic name, if it's not, call the function acquireBigIntFromBlockWindowBigIntPool or something like this